src/arrayMethodJoin.js

```diff
@@ -6,7 +6,47 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    const delimiter = getDelimiter(separator);
+
+    let output = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const current = isValidItem(this[i])
+        ? this[i]
+        : '';
```

Comment:
Can be simplified like output += isValidItem(this[i]) ? this[i] : '';

Reply:
Anyway I will fix it